### DISCUSSION TITLE
F0041

### DISCUSSION THREAD

**Discussion by adesutherland** [adesutherland/CREXX#300] [Progress Updates]
Finally (after quite a lot of weird bugs) f0041 is merged into develop

        Procedures with no arguments
        Procedures with no return value
        Line continuation
        Hex Escapes (RXAS)
        Hex/Binary literals (RXC)
        File, line and var meta data (RXC, RXAS, RXDAS)
        Allow rxbin files to be concatenated together (to create a library)
        Build changes - a consolidate library.rxbin is created in lib/rxfns directory
        Performance improvement in searching in constant pool - pointers chains for certain types, like procedures
        Added some dependencies to the cmake scripts
        Debugger PoC Version handles REXX Clauses
        Fix of register assignment logic
        Increased parser stack (100->2000) for more "if" depth. Trapped stack overflow errors
        Fix argument constant check for external procedures 
and (I forgot to add to the history file!)
        value() function

I am now going to do a quick feature to allow rexx libraries to be linked directly to the C exe

And them a long feature / basic classes and arrays

**Comment by rvjansen** [comment]
That is a lot of work!
I have merged develop into feature/f0038 to the best of my abilities - I left the head/develop messages in as comments and someone should check if I made the right choices everywhere. Most things work, I have two question marks: I added a checkBSN.rexx which fails now, and the we.rexx is 23ms in develop and 48ms in f0038. Will look into these asap.

**Comment by rvjansen** [comment]
checkBSN.rexx is fixed: it missed a return, which appears to be checked now

**Comment by adesutherland** [comment]
When would be a good time for me to eye ball the merge conflicts? Tonight?

**Comment by rvjansen** [comment]
Tonight would be ok - although we have no testcases that suddenly fail. But just to be on the safe side (and to remove the commented out sections).

**Comment by rvjansen** [comment]
Thanks! I saw the results.
